src/Microsoft.VisualStudio.Services.Agent/Capabilities/CapabilitiesManager.cs

```diff
@@ -20,17 +20,24 @@ public sealed class CapabilitiesManager : AgentService, ICapabilitiesManager
             Trace.Entering();
             ArgUtil.NotNull(settings, nameof(settings));
 
+            // Initialize a dictionary of capabilities.
+            var capabilities = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            // Skip capabilities scan for hosted agent
+            if (settings.IsProviderAgent)
```

Comment:
There is not setting that it's a provider.  providers are generic - some are single use, some static.  

The setting is whether to skip variables.  Explicit.